From 2ebaf840cd0ba9a31f6d3cc2a0fd24fb9bb36fd9 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Date: Thu, 28 Jan 2021 10:07:25 +0200
Subject: [PATCH 49/78] s32g:clk: Add SAR_ADC clock over SCMI

Replace SAR_ADC raw clock with its SCMI equivalent.

Issue: ALB-6381
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
---
 arch/arm64/boot/dts/freescale/fsl-s32-gen1.dtsi | 4 ++--
 drivers/clk/s32/s32-gen1/clk.c                  | 5 +++--
 include/dt-bindings/clock/s32gen1-scmi-clock.h  | 6 ++++--
 3 files changed, 9 insertions(+), 6 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-s32-gen1.dtsi b/arch/arm64/boot/dts/freescale/fsl-s32-gen1.dtsi
index 0555d1c5712c..287285123b93 100644
--- a/arch/arm64/boot/dts/freescale/fsl-s32-gen1.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-s32-gen1.dtsi
@@ -299,7 +299,7 @@
 		compatible = "fsl,s32gen1-adc";
 		reg = <0x0 0x401f8000 0x0 0x1000>;
 		interrupts = <0 70 4>;
-		clocks = <&clks S32GEN1_CLK_PER>;
+		clocks = <&clks S32GEN1_SCMI_CLK_SAR_ADC_BUS>;
 		clock-names = "adc";
 		vref = <1800>;
 		status = "okay";
@@ -309,7 +309,7 @@
 		compatible = "fsl,s32gen1-adc";
 		reg = <0x0 0x402e8000 0x0 0x1000>;
 		interrupts = <0 71 4>;
-		clocks = <&clks S32GEN1_CLK_PER>;
+		clocks = <&clks S32GEN1_SCMI_CLK_SAR_ADC_BUS>;
 		clock-names = "adc";
 		vref = <1800>;
 		status = "okay";
diff --git a/drivers/clk/s32/s32-gen1/clk.c b/drivers/clk/s32/s32-gen1/clk.c
index 7d79017c3ad4..84f5b23a3c2a 100644
--- a/drivers/clk/s32/s32-gen1/clk.c
+++ b/drivers/clk/s32/s32-gen1/clk.c
@@ -1,5 +1,5 @@
 /*
- * Copyright 2018,2020 NXP
+ * Copyright 2018,2020-2021 NXP
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License as published by
@@ -796,7 +796,8 @@ void s32gen1_scmi_clocks_init(void)
 		get_plat_clk(S32GEN1_CLK_XBAR);
 	scmi_clk[S32GEN1_SCMI_CLK_EDMA_AHB] =
 		get_plat_clk(S32GEN1_CLK_XBAR);
-
+	scmi_clk[S32GEN1_SCMI_CLK_SAR_ADC_BUS] =
+		get_plat_clk(S32GEN1_CLK_PER);
 
 	plat_clks.scmi_clks.clks = scmi_clk;
 	plat_clks.scmi_clks.clk_num = ARRAY_SIZE(scmi_clk);
diff --git a/include/dt-bindings/clock/s32gen1-scmi-clock.h b/include/dt-bindings/clock/s32gen1-scmi-clock.h
index 106d5cd16954..2ee06823cc1f 100644
--- a/include/dt-bindings/clock/s32gen1-scmi-clock.h
+++ b/include/dt-bindings/clock/s32gen1-scmi-clock.h
@@ -1,6 +1,6 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
 /*
- * Copyright 2020 NXP
+ * Copyright 2020-2021 NXP
  */
 #ifndef __DT_BINDINGS_SCMI_CLOCK_S32GEN1_H
 #define __DT_BINDINGS_SCMI_CLOCK_S32GEN1_H
@@ -111,7 +111,9 @@
 /* eDMA */
 #define S32GEN1_SCMI_CLK_EDMA_MODULE	S32GEN1_SCMI_CLK(67)
 #define S32GEN1_SCMI_CLK_EDMA_AHB	S32GEN1_SCMI_CLK(68)
+/* ADC */
+#define S32GEN1_SCMI_CLK_SAR_ADC_BUS	S32GEN1_SCMI_CLK(69)
 
-#define S32GEN1_SCMI_PLAT_CLK_BASE_ID	S32GEN1_SCMI_CLK(69)
+#define S32GEN1_SCMI_PLAT_CLK_BASE_ID	S32GEN1_SCMI_CLK(70)
 
 #endif
-- 
2.25.1

